backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/ArtifactHandlerBase.java

```diff
@@ -102,23 +102,19 @@ public ArtifactHandlerBase(
                           .bind("entity_id", entityId)
                           .bind("field_type", fieldType)
                           .bind("entity_name", entityName);
-                  if (!maybeKey.isPresent()) {
-                    query.bind("ar_key", maybeKey.get());
-                  }
-                  List<Artifact> artifacts =
-                      query
-                          .map(
-                              (rs, ctx) ->
-                                  Artifact.newBuilder()
-                                      .setKey(rs.getString("k"))
-                                      .setPath(rs.getString("p"))
-                                      .setArtifactTypeValue(rs.getInt("at"))
-                                      .setPathOnly(rs.getBoolean("po"))
-                                      .setLinkedArtifactId(rs.getString("lai"))
-                                      .setFilenameExtension(rs.getString("fe"))
-                                      .build())
-                          .list();
-                  return artifacts;
+                  maybeKey.ifPresent(s -> query.bind("ar_key", s));
+                  return query
+                      .map(
+                          (rs, ctx) ->
+                              Artifact.newBuilder()
+                                  .setKey(rs.getString("k"))
+                                  .setPath(rs.getString("p"))
+                                  .setArtifactTypeValue(rs.getInt("at"))
+                                  .setPathOnly(rs.getBoolean("po"))
+                                  .setLinkedArtifactId(rs.getString("lai"))
+                                  .setFilenameExtension(rs.getString("fe"))
+                                  .build())
+                      .list();
```

Comment:
Instead of getting in list and return, Directly return from here. I know this is not a bug but i have seen so improve it.